python/test/t/geometry/test_trianglemesh.py

```diff
@@ -48,3 +48,25 @@ def test_simplify_quadric_decimation():
 
     assert simplified.vertex['positions'].shape == (8, 3)
     assert simplified.triangle['indices'].shape == (12, 3)
+
+
+def test_boolean_operations():
+    box = o3d.geometry.TriangleMesh.create_box()
+    box = o3d.t.geometry.TriangleMesh.from_legacy(box)
+    sphere = o3d.geometry.TriangleMesh.create_sphere(0.8)
+    sphere = o3d.t.geometry.TriangleMesh.from_legacy(sphere)
+    # check input sphere
+    assert sphere.vertex['positions'].shape == (762, 3)
+    assert sphere.triangle['indices'].shape == (1520, 3)
+
+    ans = box.boolean_union(sphere)
+    assert ans.vertex['positions'].shape == (730, 3)
+    assert ans.triangle['indices'].shape == (1384, 3)
+
+    ans = box.boolean_intersection(sphere)
+    assert ans.vertex['positions'].shape == (154, 3)
+    assert ans.triangle['indices'].shape == (232, 3)
+
+    ans = box.boolean_difference(sphere)
+    assert ans.vertex['positions'].shape == (160, 3)
+    assert ans.triangle['indices'].shape == (244, 3)
```

Comment:
[nit] Can we check any statistical property of the triangle mesh for a more definite test?
Maybe position covariance?

Reply:
Since we use functionality from VTK we rely mostly on their tests. I think our test just detects if something has changed (VTK version update).